main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -0,0 +1,3144 @@
+        case (ident, types, Some(latTerm)) => PredicateParam.PredicateParamWithType(Name.mkPred(ident), Denotation.Latticenal, types :+ latTerm, tree.loc)
+      }
+    }
+
+    private def tryPickLatticeTermExpr(tree: Tree)(implicit s: State): Validation[Option[Expr], CompilationMessage] = {
+      traverseOpt(tryPick(TreeKind.Predicate.LatticeTerm, tree))(Exprs.pickExpression)
+    }
+
+    private def tryPickLatticeTermType(tree: Tree)(implicit s: State): Validation[Option[Type], CompilationMessage] = {
+      traverseOpt(tryPick(TreeKind.Predicate.LatticeTerm, tree))(Types.pickType)
+    }
+
+    private def tryPickLatticeTermPattern(tree: Tree)(implicit s: State): Validation[Option[Pattern], CompilationMessage] = {
+      traverseOpt(tryPick(TreeKind.Predicate.LatticeTerm, tree))(Patterns.pickPattern(_))
+    }
+
+  }
+
+  private object Types {
+    def pickType(tree: Tree)(implicit s: State): Validation[Type, CompilationMessage] = {
+      flatMapN(pick(TreeKind.Type.Type, tree))(visitType)
+    }
+
+    def tryPickTypeNoWild(tree: Tree)(implicit s: State): Validation[Option[Type], CompilationMessage] = {
+      mapN(pickType(tree)) {
+        case Type.Var(ident, _) if ident.isWild => None
+        case t => Some(t)
+      }
+    }
+
+    def tryPickType(tree: Tree)(implicit s: State): Validation[Option[Type], CompilationMessage] = {
+      val maybeType = tryPick(TreeKind.Type.Type, tree)
+      traverseOpt(maybeType)(visitType)
+    }
+
+    def visitType(tree: Tree)(implicit s: State): Validation[WeededAst.Type, CompilationMessage] = {
+      assert(tree.kind == TreeKind.Type.Type)
+      // Visit first child and match its kind to know what to to
+      val inner = unfold(tree)
+      inner.kind match {
+        case TreeKind.Type.Ascribe => visitAscribe(inner)
+        case TreeKind.Type.Variable => visitVariable(inner)
+        case TreeKind.Type.Binary => visitBinary(inner)
+        case TreeKind.Type.Unary => visitUnary(inner)
+        case TreeKind.Type.Apply => visitApply(inner)
+        case TreeKind.Type.Tuple => visitTuple(inner)
+        case TreeKind.Type.Native => visitNative(inner)
+        case TreeKind.Type.CaseSet => visitCaseSet(inner)
+        case TreeKind.Type.Record => visitRecord(inner)
+        case TreeKind.Type.RecordRow => visitRecordRow(inner)
+        case TreeKind.Type.Schema => visitSchema(inner)
+        case TreeKind.Type.SchemaRow => visitSchemaRow(inner)
+        case TreeKind.Type.Constant => visitConstant(inner)
+        case TreeKind.Type.EffectSet => visitEffect(inner)
+        case TreeKind.QName => visitName(inner)
+        case TreeKind.Ident => mapN(tokenToIdent(inner)) {
+          case ident if ident.isWild => Type.Var(ident, tree.loc)
+          case ident => Type.Ambiguous(Name.mkQName(ident), inner.loc)
+        }
+        // TODO: This double reports the same error. We should be able to handle this resiliently if we had Type.Error.
```

Comment:
We do have Type.Error, no?

Reply:
I don't think so. I wasn't able to find it

Reply:
We have it, at least later in the compiler. https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/ast/TypeConstructor.scala#L417

If you need a WeededAst.Type.Error, just add it  at the bottom :)

Reply:
I will add it.

Reply:
https://github.com/flix/flix/pull/7469